File: src/main/java/br/com/matheuspadilha/tasks/dto/TaskDTO.java
package br.com.matheuspadilha.tasks.dto;

import br.com.matheuspadilha.tasks.enums.TaskState;
import br.com.matheuspadilha.tasks.model.Address;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDate;

import static com.fasterxml.jackson.annotation.JsonInclude.Include;

@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonInclude(Include.NON_NULL)
public class TaskDTO implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    private String id;
    private String title;
    private String description;
    private Integer priority;
    private TaskState state;
    private Address address;
    private LocalDate created;

}

File: src/main/kotlin/br/com/matheuspadilha/tasks/dto/TaskDTO.kt
package br.com.matheuspadilha.tasks.dto

import br.com.matheuspadilha.tasks.enums.TaskState
import br.com.matheuspadilha.tasks.model.Address
import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonInclude.Include
import java.io.Serializable
import java.time.LocalDate

@JsonInclude(Include.NON_NULL)
data class TaskDTO (
    val id: String? = null,
    val title: String? = null,
    val description: String? = null,
    val priority: Int,
    val state: TaskState? = null,
    val address: Address? = null,
    val created: LocalDate? = null
) : Serializable
